### PR TITLE
Fix eslint warnings

### PR DESCRIPTION
**Pull request by matheusps**
#### What is the purpose of this pull request?

As the title says

#### What problem is this solving?

Respect eslint warnings can prevent bugs.

#### How should this be manually tested?

`yarn && yarn lint`

#### Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/EXPERIMENTAL_useCheckboxTree/constants.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
-export const defaultComparatorCurry = (item: any) => (candidate: any) =>
-  item.id === candidate.id
+type itemWithID = {
```
Comment (rerissondaniel) [inline react/components/EXPERIMENTAL_useCheckboxTree/constants.ts]:
```suggestion
type ItemWithID = {
```
(But just suggestion)

File: react/components/EXPERIMENTAL_Charts/helpers.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ const merge = (defaultProps: any, userProps: any) => {
   return props
 }
 
-const getRangeOfZAxis = (key, data) => {
+const getRangeOfZAxis = (key: string | number, data: unknown[]) => {
```
Comment (rerissondaniel) [inline react/components/EXPERIMENTAL_Charts/helpers.ts]:
```suggestion
const getRangeOfZAxis = (key: string | number, data: object[]) => {
```
To be consistent with `typings.d.ts`?

File: react/components/EXPERIMENTAL_useCheckboxTree/constants.ts (hunk 1 above)
Comment (matheusps) [reply react/components/EXPERIMENTAL_useCheckboxTree/constants.ts]: Accepted

File: react/components/EXPERIMENTAL_Charts/helpers.ts (hunk 2 above)
Comment (matheusps) [reply react/components/EXPERIMENTAL_Charts/helpers.ts]: Sure

**Pull request merged**